dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ExecutorController.java

```diff
@@ -138,6 +139,100 @@ public Result startProcessInstance(@ApiIgnore @RequestAttribute(value = Constant
+                                       @RequestParam(value = "taskDependType", required = false) TaskDependType taskDependType,
+                                       @RequestParam(value = "execType", required = false) CommandType execType,
+                                       @RequestParam(value = "warningType", required = true) WarningType warningType,
+                                       @RequestParam(value = "warningGroupId", required = false) int warningGroupId,
+                                       @RequestParam(value = "runMode", required = false) RunMode runMode,
+                                       @RequestParam(value = "processInstancePriority", required = false) Priority processInstancePriority,
+                                       @RequestParam(value = "workerGroup", required = false, defaultValue = "default") String workerGroup,
+                                       @RequestParam(value = "environmentCode", required = false, defaultValue = "-1") Long environmentCode,
+                                       @RequestParam(value = "timeout", required = false) Integer timeout,
+                                       @RequestParam(value = "startParams", required = false) String startParams,
+                                       @RequestParam(value = "expectedParallelismNumber", required = false) Integer expectedParallelismNumber,
+                                       @RequestParam(value = "dryRun", defaultValue = "0", required = false) int dryRun) {
+
+        if (timeout == null) {
+            timeout = Constants.MAX_TASK_TIMEOUT;
+        }
+        Map<String, String> startParamMap = null;
+        if (startParams != null) {
+            startParamMap = JSONUtils.toMap(startParams);
+        }
+
+        Map<String, Object> result = new HashMap<>();
+        String[] processDefinitionCodeArray = processDefinitionCodes.split(",");
+        Set<Long> processDefinitionCodeSet = new HashSet<>();
+        List<String> startFailedProcessDefinitionCodeList = new ArrayList<>();
+
+        for (String StrProcessDefinitionCode :processDefinitionCodeArray) {
+            long processDefinitionCode = Long.parseLong(StrProcessDefinitionCode);
+            processDefinitionCodeSet.add(processDefinitionCode);
+        }
+
+        for (long processDefinitionCode : processDefinitionCodeSet) {
+            result = execService.execProcessInstance(loginUser, projectCode, processDefinitionCode, scheduleTime, execType, failureStrategy,
+                    startNodeList, taskDependType, warningType, warningGroupId, runMode, processInstancePriority, workerGroup, environmentCode, timeout, startParamMap, expectedParallelismNumber, dryRun);
+
+            if (!Status.SUCCESS.equals(result.get(Constants.STATUS))) {
+                startFailedProcessDefinitionCodeList.add(String.valueOf(processDefinitionCode));
+            }
```

Comment:
I have a question, should we termination all other start if any of `processDefinitionCodeSet` failed

Reply:
I think we should not terminating all other start if any of ```processDefinitionCodeSet``` failed in case of process definition offline or invalid process definition code.

Reply:
It's Ok, I don't have a strong opinion here, but we should add our strategy to java doc or something else.

Reply:
OK.I'll add it to java doc.